app/models/UserAnswers.scala

```diff
@@ -30,20 +31,27 @@ final case class UserAnswers(internalId: String,
                              identifier: String,
                              data: JsObject = Json.obj(),
                              is5mldEnabled: Boolean = false,
-                             isTrustTaxable: Boolean = true,
+                             isUnderlyingData5mld: Boolean = false,
                              updatedAt: LocalDateTime = LocalDateTime.now) extends Logging {
 
   def identifierType: IdentifierType = if (identifier.matches(Validation.utrRegex)) UTR else URN
 
-  def trustMldStatus: TrustMldStatus = (is5mldEnabled, isUnderlyingTrust5mld, isTrustTaxable) match {
+  def trustTaxability: TrustTaxability = (this.get(WhatIsNextPage), identifierType) match {
+    case (Some(NeedsToPayTax), _) => MigratingFromNonTaxableToTaxable
+    case (Some(NoLongerTaxable), _) => MigratingFromTaxableToNonTaxable
+    case (_, UTR) => Taxable
+    case (_, URN) => NonTaxable
```

Comment:
I'm not sure we're safe in assuming they're taxable if they have a UTR in all cases. May be one to speak to Wendy about.